### ISSUE TITLE
Telescope Control Enhancements for Satellite Tracking

### ISSUE THREAD

**Issue by rkinnett** [Stellarium/stellarium#2361]
The telescope control interface operates on static coordinates and snapshots of an object's coordinates in time via the "Current object" button.  Fast-moving objects like comets and satellites quickly drift away from the position captured with the "Current object" button.  That said, it is possible to crudely track even fast-moving LEO satellites by rapidly clicking "Current object" and Slew, but this interaction can be easily improved and should be useful to myriad applications.

Please consider adding a checkbox or similar UI to the "Slew telescope to" GUI to automatically update RA and Dec coordinates in the GUI as a selected target moves or the user selects different targets.

Please also consider adding a separate button or checkbox near the Slew button to periodically transmit selected target coordinates to a telescope mount.  This may be more complicated since there are practical limits to how quickly telescopes mounts can accommodate successive go-to commands, but perhaps the user can manage that if provided some way to specify a telescope command period?

Lastly, please consider augmenting the remote telescope control TCP protocol to include a satellite information message (minimally including NORAD ID and International Designator, if not whole TLE) and add a button to the telescope control GUI to send that over TCP.  I suppose this feature might be uniquely useful to external software, not telescope mounts in general, so adding a button to the telescope control GUI would be awkward.

I very much appreciate all of the fine work on this project and am enjoying integrating Stellarium with my satellite tracking software.  I have implemented the Stellarium telescope control TCP protocol to send telescope pointing coordinates to Stellarium and to receive static go-to commands from Stellarium.  The last detail needed for complete integration is a way to send satellite ID to the tracker application.

Thank you for considering.

**Comment by rkinnett** [comment]
Slight update, just a button "Send TLE to external application" (along with IP address and port inputs in settings) would be tremendously helpful.  I have coordinated with the author of SkyTrack and am reaching out to other planetarium software devs to hopefully make this a common utility for linking satellite tracking applications.  Would love to see Stellarium help us build this capability!  Currently we just pass TLEs from a graphical application to a satellite tracking application, with no telemetry feedback, but we could discuss establishing a standard API.  Thanks again for considering.

**Comment by alex-w** [comment]
This is a good task for the community to participate in the contribution into Stellarium. Who wants to help us?

**Comment by microp11** [comment]
Hi, not sure if this would be to help to anyone else. About 2 years ago I have modified the satellite plugin (and the telescope protocol - as it was the way I found it easiest) to pass the AZ and EL of the satellite to control a small rotor controlled by an Arduino. About a week ago I started writing another small application that emulates the Orbitron DDE interface to control a SPID rotor for a bigger dish. The code for both is included in their respective repositories:

https://github.com/microp11/stellarium
https://github.com/microp11/Telescope

And adding a picture of both the modified telescope client and the telescope and DDE servers plus the SPID controller.
In hope that this could help someone else.

![image](https://user-images.githubusercontent.com/1545798/186289214-0214a123-e33b-4453-ae35-1af6f942b743.png)

**Comment by gzotti** [comment]
Looks at least interesting! We had some requests to also support GOTO in alt-azimuth telescopes. Not sure what it takes, I don't run such hardware, but we usually accept pull requests of working and well-documented solutions which at least don't break existing setups. User documentation goes into the Stellarium User Guide and should be complete enough to be understood by average amateur astronomers with some inclination to hardware tinkering (if that is necessary. I see Arduino & friends...).  

If the Alt/Az autoslew is only available for particular telescopes, this must be documented, and the GUI elements hidden for instruments which don't have this feature supported. Probably for most equatorially mounted telescopes, autoslew in equatorial coordinates could be added as well. And yes, some safety net for too slow motors, or stop of tracking when meridian flip is required for GE mounts. And a big red STOP button if something goes very wrong...

**Comment by gzotti** [comment]
The Satellites plugin can only track objects in (moderately close) Earth orbit. We had issues with users wanting to track the JWST. Artemis or other Lunar missions will not work after the zone of "common" TLE.

**Comment by microp11** [comment]
My understanding is as well that the TLE describe an elliptical orbit around Earth. See here same problem and some answers: https://space.stackexchange.com/questions/57771/does-the-james-webb-telescope-have-a-two-line-element-set

**Comment by rkinnett** [comment]
Right, that’s what I meant by not being being able to track Orion much longer.. as soon as it reaches the Moon’s sphere of influence, the current TLE will be invalid and the lunar orbit cannot be described by TLE.

Which leads me to another motivation for wanting to learn how to build plugins from source.  I want to also make a plugin to interface with JPL’s Horizons ephemeris system to track deep space objects beyond Earth orbit.  A topic for another day..

**Comment by kefeur** [comment]
HI all! 
I downloaded version 24.2, but I can't find the tracker module in it. There is no orbitron dde in it either. Is there a downloadable version that includes it? I want satellite tracking.

**Comment by microp11** [comment]
> HI all! I downloaded version 24.2, but I can't find the tracker module in it. There is no orbitron dde in it either. Is there a downloadable version that includes it? I want satellite tracking.

The Orbitron DDE is part of the Orbitron software.
